src/profile-logic/zip-files.js

```diff
@@ -195,8 +195,17 @@ export class ZipFileTree {
           encodeURIComponent(this._zipFileTable.path[zipTableIndex]);
       }
 
+      let name = this._zipFileTable.partName[zipTableIndex];
+      const EXTENSIONS_TO_OMIT = ['.json', '.json.gz'];
+      const matchedExtension = EXTENSIONS_TO_OMIT.find((extension) =>
+        name.endsWith(extension)
+      );
+      if (matchedExtension) {
+        name = name.slice(0, name.length - matchedExtension.length);
+      }
```

Comment:
I would have done it with a regexp such as:
```js
let name = ...
const extensionsToOmitRe = /\.json(\.gz)?$/;
name = name.replace(extensionsToOmitRe, "");
```
It's terser but also maybe less explicit for some people.

Your solution is valid too, and I don't think one is better than the other in most cases, so I'll let you decide which one you prefer (just in case you didn't think of the RegExp based solution).

Reply:
Thanks for the suggestion!

I tend to lean away from using regex for most cases, so if possible I'd prefer to keep it as is